src/command_modules/azure-cli-component/azure/cli/command_modules/component/custom.py

```diff
@@ -16,6 +16,10 @@
 COMPONENT_PREFIX = 'azure-cli-'
 
 
+def _deprecate_warning():
+    logger.warning("The 'component' commands will be deprecated in the future.")
```

Comment:
The pattern for deprecating stuff is to register a command parsing callback, like https://github.com/Azure/azure-cli/blob/master/src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/commands.py#L21

Hope this saves you a few code change, but up to you.